backend/dal/Migrations/Initial/PostDeploy/01-ProjectStatus.sql

```diff
@@ -6,70 +6,94 @@ INSERT INTO dbo.[ProjectStatus] (
     , [Name]
     , [IsDisabled]
     , [Description]
+    , [Route]
+    , [Workflow]
 ) VALUES (
     0
     , 0
     , 'Draft'
     , 0
     , 'A new draft project that is not ready to submit to apply to be added to the Surplus Property Program.'
+    , '/project/draft'
```

Comment:
These `routes` will need to exist in some format in the APP.

Reply:
My initial reaction would be to question why would the backend need to dictate a client-side routing concern? But I'd love to learn more what's the rationale behind this change?

Reply:
It was an option.  Either the APP maintains constants that link to the primary keys for each project status, or the db has a way to configure what each step says and does (limited).  We choose the db configurable way.  Now the APP makes a request to an endpoint to find out what the workflow is and uses it to move from page to page.  It of course isn't fully configurable, but it does allow the APP to do less hard-coding.

Reply:
I recognize this is a first stab at designing a "workflow engine" on the backend and am happy to see how it works with this iteration. However, I'd like to propose that we discuss the design at a later time (not too late).

About the APP having to keep constants I'm not sure that's necessary as we already have a way to get lookup values from the server. We could have an endpoint like:

```
GET /workflows/{id}/available-transitions
```

That returns those constants - say for our first workflow "SubmitDisposal" (e.g. id = 1)

```
GET /workflows/{id}/available-transitions

# response

{
  transitions: [
    {
       id: 1,
       name: "Draft",
       order: 1  // first step
    },
    {
      id: 2,
      name: "Select Properties"
      order: 2  // second step
    },
    ....
  ]
}
```

Reply:
Or we can do a nearly identical solution which is what we're doing `ProjectStatus` = a list of valid transitions that a project can go through.

Reply:
I'm no expert in designing workflow engines but a couple google searches yield good resources:

- https://www.kennethlange.com/how-to-model-workflows-in-rest-apis/
- http://joyent.github.io/node-workflow/workflowapi.html

Reply:
We're not looking for an elaborate workflow engine, simply an array of steps/stages/statuses that we can configure with small details (name, description, order, route).  This reduces the requirement to hard-code primary keys into the APP.

Reply:
Additionally your first link 3rd example is essentially what is here.